wal/wal_test.go

```diff
@@ -547,6 +548,121 @@ func TestWAL_Repair(t *testing.T) {
 	}
 }
 
+func TestCorruptAndCarryOn(t *testing.T) {
+	dir, err := ioutil.TempDir("", "wal_repair")
+	testutil.Ok(t, err)
+	defer os.RemoveAll(dir)
+
+	logger := testutil.NewLogger(t)
+
+	// Produce a WAL with a two segments of 3 pages with 3 records each,
+	// so when we truncate the file we guaranteed to split a record.
+	{
+		w, err := NewSize(logger, nil, dir, pageSize*3)
```

Comment:
can set this as a var and use it few places
`expWalSize := int64(pageSize * 3)`